File: app/src/main/java/com/supercilex/robotscouter/ui/teamlist/NewTeamDialog.java
package com.supercilex.robotscouter.ui.teamlist;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.TextInputLayout;
import android.support.v4.app.FragmentManager;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;

import com.supercilex.robotscouter.R;
import com.supercilex.robotscouter.data.model.Team;
import com.supercilex.robotscouter.ui.KeyboardDialogBase;

public class NewTeamDialog extends KeyboardDialogBase {
    private static final String TAG = "NewTeamDialog";

    private TextInputLayout mInputLayout;
    private EditText mTeamNumberEditText;

    public static void show(FragmentManager manager) {
        new NewTeamDialog().show(manager, TAG);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        View rootView = View.inflate(getContext(), R.layout.dialog_new_team, null);
        mInputLayout = (TextInputLayout) rootView.findViewById(R.id.name);
        mTeamNumberEditText = (EditText) mInputLayout.findViewById(R.id.team_number);
        return createDialog(rootView, R.string.add_scout);
    }

    @Override
    protected EditText getLastEditText() {
        return mTeamNumberEditText;
    }

    @Override
    public boolean onClick() {
        String teamNumber = mTeamNumberEditText.getText().toString();
        if (isValid(teamNumber)) {
            teamNumber = String.valueOf(Long.parseLong(teamNumber));
            ((TeamSelectionListener) getActivity())
                    .onTeamSelected(new Team.Builder(teamNumber).build(), true);
            return true;
        } else {
            mInputLayout.setError(getString(R.string.invalid_team_number));
            return false;
        }
    }

    private boolean isValid(String teamNumber) {
        if (TextUtils.isEmpty(teamNumber)) return false;

        try {
            long number = Long.parseLong(teamNumber);
            return number >= 0 && number <= 100000;
        } catch (NumberFormatException e) {
            return false;
        }
    }
}

File: app/src/main/java/com/supercilex/robotscouter/ui/teamlist/NewTeamDialog.kt
package com.supercilex.robotscouter.ui.teamlist

import android.os.Bundle
import android.support.design.widget.TextInputEditText
import android.support.v4.app.FragmentManager
import android.view.View
import com.supercilex.robotscouter.R
import com.supercilex.robotscouter.data.model.Team
import com.supercilex.robotscouter.util.data.getScoutBundle
import com.supercilex.robotscouter.util.ui.KeyboardDialogBase
import com.supercilex.robotscouter.util.ui.TeamSelectionListener
import com.supercilex.robotscouter.util.unsafeLazy
import kotlinx.android.synthetic.main.dialog_new_team.*

class NewTeamDialog : KeyboardDialogBase() {
    override val containerView: View by unsafeLazy {
        View.inflate(context, R.layout.dialog_new_team, null)
    }
    override val lastEditText: TextInputEditText by unsafeLazy { number }

    override fun onCreateDialog(savedInstanceState: Bundle?) =
            createDialog(R.string.scout_new_title, savedInstanceState)

    public override fun onAttemptDismiss(): Boolean {
        val teamNumber = try {
            lastEditText.text.toString().toLong()
        } catch (e: NumberFormatException) {
            name.error = getString(R.string.number_too_big_error)
            return false
        }
        (activity as TeamSelectionListener)
                .onTeamSelected(getScoutBundle(Team(teamNumber, ""), true))
        return true
    }

    companion object {
        private const val TAG = "NewTeamDialog"

        fun show(manager: FragmentManager) = NewTeamDialog().show(manager, TAG)
    }
}
